### ISSUE TITLE
Redesign DB schema (allows for duplicate keys by means of different field order as is)

### ISSUE THREAD

**Issue by rgreschner** [rgreschner/mongodb-iot-mqtt-example#26]
As pointed out by @alanderex , it is currently possible to mix-up ids of sensor data because of field order stored in compound index.
E.g. when inserting two documents

```
db.idTest.insert({"_id":{x:1, y:0}})
```

and

```
db.idTest.insert({"_id":{y:0, x:1}})
```

actually 2 documents get created instead of one. This also affects queries.

Problem can be mitigated by creating unique constraint on id fields, but since this seems to be a general problem with some drivers (e.g. Python), a flattening of compund index fields into collection fields and the introduction of unique constraints on them seems more reasonable.

**Comment by rgreschner** [comment]
Fixed in commit b5874581b69fcbf9610cd09481a901e45def8080.
Unique constraint on fields is probably not necessary, also it would restrict amount of simultaneously loggable data at the exact same moment (assumption in the past was that values are distributed with larger gaps in time)..

**Comment by alanderex** [comment]
+1
